### PR TITLE
feat: edit profile picture

### LINKED ISSUE #1214: User cannot edit Profile Picture
**Summary:**

User cannot edit Profile Picture in User Details fragment. Nothing happens when Edit Floating Action Button is pressed.

**Steps to reproduce:**

Open App -> Click on the profile pic -> Click on Edit Icon (nothing happens)

**Expected behaviour:**

The user should be able to change his profile picture.

**Observed behaviour:**

Nothing happens when clicked on edit icon

**Device and Android version:**

Android 10
OpenPlus 6 
Stock Rom

### PR DESCRIPTION
**Pull request by laxyapahuja**
Fixes #1214 

ezgif-4-696ef9f79ec3

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java [hunk 1]
```diff
@@ -157,6 +178,66 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
+                cursor.close();
+                // Set the Image in ImageView after decoding the String
+                ivProfile.setVisibility(View.GONE);
+                ivTextDrawable.setVisibility(View.GONE);
+                iv_profilepic.setVisibility(View.VISIBLE);
+                iv_profilepic.setImageBitmap(BitmapFactory
+                        .decodeFile(imgDecodableString));
+
+            } else {
+                Toast.makeText(getActivity(), "You haven't picked Image",
```
Comment (luckyman20) [inline app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java]: Use `Toaster` library that we have in utils.

Comment (laxyapahuja) [reply app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java]: done 😃

**Comment by laxyapahuja** [comment]
@elirehema @miPlodder hey, can you please review the task? thanks :)

**Comment by luckyman20** [comment]
@laxyapahuja Change commit title to `feat: edit proile picture` and clear up the commit message.

**Comment by laxyapahuja** [comment]
> @laxyapahuja Change commit title to `feat: edit proile picture` and clear up the commit message.

@luckyman20 done :D

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -567,6 +567,8 @@
     <string name="savings_account_transaction">Savings Account Transaction</string>
     <string name="transaction_period">Transaction Period</string>
     <string name="transaction_type">Transaction Type</string>
+    <string name="notpickedimage">You haven\'t picked image.</string>
```
Comment (luckyman20) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="image_not_picked">You haven\'t picked image.</string>
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -567,6 +567,8 @@
...
+    <string name="somethingwentwrong">Something went wrong.</string>
```
Comment (luckyman20) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="something_went_wrong">Something went wrong.</string>
```

**Review by luckyman20 — CHANGES_REQUESTED** [review]
Change the java code accordingly.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (laxyapahuja) [reply app/src/main/res/values/strings.xml]: @luckyman20 done :)

File: app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java [hunk 4]
```diff
@@ -157,6 +177,64 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
+
+    private void checkPermissions() {
+        if (ContextCompat.checkSelfPermission(getActivity(),
+                Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(getActivity(), new String[]{
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE},
+                    1001);
+        }   else {
+            int i = 1;
```
Comment (luckyman20) [inline app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java]: @laxyapahuja is this used anywhere? what is its purpose?

Comment (laxyapahuja) [reply app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java]: no, i used it as a workaround 😄 i've removed it in the latest commit.

**Pull request closed without merging**